sssd_test_framework/utils/smartcard.py

```diff
@@ -0,0 +1,194 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from pytest_mh import MultihostHost, MultihostUtility
+from pytest_mh.cli import CLIBuilder, CLIBuilderArgs
+from pytest_mh.utils.fs import LinuxFileSystem
+from pytest_mh.utils.services import SystemdServices
+
+if TYPE_CHECKING:
+    from ..roles.client import Client
+
+__all__ = [
+    "SmartCardUtils",
+]
+
+
+class SmartCardUtils(MultihostUtility[MultihostHost]):
+    """
+    Utility class for managing smart card operations using SoftHSM and PKCS#11.
+    """
+
+    SOFTHSM2_CONF_PATH = "/opt/test_ca/softhsm2.conf"
+    TOKEN_STORAGE_PATH = "/opt/test_ca/tokens"
+    OPENSC_CACHE_PATHS = [
+        "$HOME/.cache/opensc/",
+        "/run/sssd/.cache/opensc/",
+    ]
+
+    def __init__(self, host: MultihostHost, fs: LinuxFileSystem, svc: SystemdServices) -> None:
+        """
+        :param host: Multihost host.
+        :type host: MultihostHost
+        :param fs: Filesystem utility instance.
+        :type fs: LinuxFileSystem
+        :param svc: Systemd svc utility instance.
+        :type svc: SystemdServices
+        """
+        super().__init__(host)
+
+        self.cli: CLIBuilder = host.cli
+        """CLI builder utility for command construction."""
+
+        self.fs: LinuxFileSystem = fs
+        """Filesystem utility used to handle file operations."""
+
+        self.svc: SystemdServices = svc
+        """Systemd utility to manage and interact with svc."""
+
+    def create_certs(self, label: str = "sc_test", so_pin: str = "12345678", user_pin: str = "123456") -> None:
+        """
+        Initializes a SoftHSM token with the given label and PINs.
```

Comment:
I think this method is poorly named. I'll need @spoore1 to time in. 

It's called ```create_certs```, so it should create the public/private key? Does ```initialize_card```, ```provision_card``` a better description of what this method does?